core/src/avm2/globals/flash/net/url_loader.rs

```diff
@@ -0,0 +1,175 @@
+//! `flash.net.URLLoader` builtin/prototype
+
+use crate::avm2::activation::Activation;
+use crate::avm2::class::Class;
+use crate::avm2::method::{Method, NativeMethodImpl};
+use crate::avm2::names::{Namespace, QName};
+use crate::avm2::object::TObject;
+use crate::avm2::value::Value;
+use crate::avm2::{Error, Object};
+use crate::backend::navigator::RequestOptions;
+use crate::loader::DataFormat;
+use crate::string::AvmString;
+use gc_arena::{GcCell, MutationContext};
+
+/// Implements `flash.net.URLLoader`'s class constructor.
+pub fn class_init<'gc>(
+    _activation: &mut Activation<'_, 'gc, '_>,
+    _this: Option<Object<'gc>>,
+    _args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error> {
+    Ok(Value::Undefined)
+}
+
+/// Implements `flash.net.URLLoader`'s instance constructor.
+pub fn instance_init<'gc>(
+    activation: &mut Activation<'_, 'gc, '_>,
+    this: Option<Object<'gc>>,
+    args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error> {
+    if let Some(mut this) = this {
+        activation.super_init(this, &[])?;
+        this.set_property(
+            &QName::new(Namespace::public(), "dataFormat").into(),
+            "text".into(),
+            activation,
+        )?;
+        this.set_property(
+            &QName::new(Namespace::public(), "data").into(),
```

Comment:
uninitialized slot should be `undefined` by default, right? No need to set it manually, I think?

Reply:
It appeared to be `null` when I tested it

Reply:
that's weird, looking at the code we should be default-initializing these to Undefined... I'll check later too

Reply:
This happens here, where we create a slot trait with no default value: https://github.com/ruffle-rs/ruffle/blob/de99dcd37b5442a87c011d603858811a1947aecd/core/src/avm2/class.rs#L755-L759

This results in us picking `null` as the default slot value: https://github.com/ruffle-rs/ruffle/blob/de99dcd37b5442a87c011d603858811a1947aecd/core/src/avm2/traits.rs#L405

Side node - is the "slot type" actually used anywhere besides picking the default value?

Reply:
> Side node - is the "slot type" actually used anywhere besides picking the default value?

Pretty sure it should be used for type checks, implicit conversions? And JIT ofc

> This results in us picking null as the default slot value:

Oh, I see now, it's here:

https://github.com/ruffle-rs/ruffle/blob/de99dcd37b5442a87c011d603858811a1947aecd/core/src/avm2/traits.rs#L159

For some reason I thought the `None` gets propagated all the way to actual slot value iniitialization, up to here:
https://github.com/ruffle-rs/ruffle/blob/de99dcd37b5442a87c011d603858811a1947aecd/core/src/avm2/object/script_object.rs#L294-L298

But you're right, it's indeed `null` by default in FP.